docs/content/wiring.md

```diff
@@ -0,0 +1,321 @@
+---
+title: Using Parameters, Credentials and Outputs
+description: How to wire parameters, credentials and outputs into steps
+---
+
+# Parameters, Credentials and Outputs in Porter
+
+In the Porter manifest, you can declare both [parameters](https://github.com/deislabs/cnab-spec/blob/master/101-bundle-json.md#parameters) and [credentials](https://github.com/deislabs/cnab-spec/blob/master/101-bundle-json.md#credentials), which are defined in the CNAB spec. The CNAB specification specifies how both credentials and parameters can be provided to an [invocation image](https://github.com/deislabs/cnab-spec/blob/master/102-invocation-image.md) and how they should be declared in a bundle manifest. 
+
+The Porter manifest allows you to provide all the elements defined in the specification, but with Porter's opinionated approach by default. In addition to providing a mechanism for declaring parameters and credentials at the bundle level, Porter provides a way to declare how each of these are provided to [mixins](/mixin-architecture). This mechanism is also applicable to declaring how output from one mixin can be passed to another, as well as how to consume parameters, credentials and outputs from bundle dependencies.
+
+## Parameters
+
+In order to declare a parameter in a Porter bundle, you first declare a parameters block with one or more parameters in the `porter.yaml`. For example, to declare a parameter named database_name, you might include the following YAML block:
+
+```yaml
+parameters:
+- name: database_name
+  type: string
+```
+
+This is the minimum required to create a parameter in Porter. Porter will specify an environment variable destination that defaults to the upper-cased name of the parameter. This would result in the following parameter definition in the `bundle.json`:
+
+```json
+"parameters": {
+    "database_name": {
+        "type": "string",
+        "required": false,
+        "metadata": {},
+        "destination": {
+            "path": "",
+            "env": "DATABASE_NAME"
+        }
+    }
+}
+```
+
+You can also provide any other attributes, as specified by the CNAB [parameters](https://github.com/deislabs/cnab-spec/blob/master/101-bundle-json.md#parameters) specification. To specify a default value, for example, you could provide the following parameter definition:
+
+```yaml
+- name: database_name
+  type: string
+  default: "wordpress"
+```
+
+This would result in the following `bundle.json` parameter definition:
```

Comment:
Is it useful to spell out in the doc here how it gets turned into parameter in bundle.json? We don't do that for anything else.

Reply:
Removed.